substrate/utils/wasm-builder/src/wasm_project.rs

```diff
@@ -711,22 +764,34 @@ fn offline_build() -> bool {
 
 /// Build the project and create the bloaty runtime blob.
 fn build_bloaty_blob(
+	target: RuntimeTarget,
 	blob_build_profile: &Profile,
 	project: &Path,
 	default_rustflags: &str,
 	cargo_cmd: CargoCommandVersioned,
-) {
+) -> PathBuf {
 	let manifest_path = project.join("Cargo.toml");
 	let mut build_cmd = cargo_cmd.command();
 
-	let rustflags = format!(
-		"-C target-cpu=mvp -C target-feature=-sign-ext -C link-arg=--export-table {} {}",
-		default_rustflags,
-		env::var(crate::WASM_BUILD_RUSTFLAGS_ENV).unwrap_or_default(),
-	);
+	let mut rustflags = String::new();
+	match target {
+		RuntimeTarget::Wasm => {
+			rustflags.push_str(
+				"-C target-cpu=mvp -C target-feature=-sign-ext -C link-arg=--export-table ",
+			);
+		},
+		RuntimeTarget::Riscv => {
+			rustflags.push_str("-C target-feature=+lui-addi-fusion -C relocation-model=pie -C link-arg=--emit-relocs -C link-arg=--unique ");
```

Comment:
I was hitting duplicate symbol name when implementing PolkaVM support for ink!. Essentially the `call` export was having problems. Is the `--unique` a solution for that?

Reply:
No, the `--unique` just allows LLD to emit multiple sections, which allows our linker to better optimize the blob in some cases.

Your problems with duplicate symbols are probably fixed in the newest version of the PolkaVM derive crate. With each successive version I've been coming up with less hacky ways to do imports/exports, and the way they're implemented right now manages to exclusively use `rustc` to generate the symbol names for us so there should not be any dupes anymore.